### PR TITLE
Make the cluster federation bootstrap scripts a little bit more readable so it is easier to debug.

### PR DESCRIPTION
**Pull request by madhusudancs**
Only the last commit here needs review.

cc @colhom @kubernetes/sig-cluster-federation

[Analytics]()

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/26950)

### CODE REVIEW

**Comment by colhom** [comment]
@madhusudancs most of this PR is shuffling indent spacing. I made an effort to have the bash indent scheme that's there match "most" of the rest of the repo. Is that not the case?

**Comment by madhusudancs** [comment]
@colhom you are right, shuffling indentation is mostly what this PR is doing and that's the intention as well. I see no common convention in our bash scripts. For example, [kube-util.sh](https://github.com/kubernetes/kubernetes/blob/a092fd2dd4091b735cc2389015ce4bef951abfab/cluster/kube-util.sh) uses tabs, but [gce/util.sh](https://github.com/kubernetes/kubernetes/blob/7476d97781563b70e8b89a8bd3f99ea75ae6c290/cluster/gce/util.sh) and [lib/golang.sh](https://github.com/kubernetes/kubernetes/blob/7476d97781563b70e8b89a8bd3f99ea75ae6c290/hack/lib/golang.sh) use 2-spaces. 

What I am trying to do is removing the tab-space mixup in a single file, i.e. bring tab or space uniformity. Mixing up them up is painful in some text editors. This PR is not 100% there yet, but this is a first step so at least I can read/debug the scripts. My eventual goal is to make them the federation scripts at least 2-spaced.

**Comment by colhom** [comment]
@madhusudancs i think that if you're finding the bash code painful to read, we should look into applying a universal format to all the bash scripts in the repo. There's no point in doing it piecemeal and interspersed with functional changes to the code-base.  It would be huge PR- but also entirely mechanical operation.

We could then just have a bash lint detector as part of the submit queue checks.

**Comment by madhusudancs** [comment]
@colhom I agree with you. There are two things specific to this PR.

1. Intentionally and very carefully, this PR doesn't make any functional changes. Particularly to keep for the reason you mention. I understand that you are proposing to make the changes in one go, but read the next point.
2. `federation/cluster/common.sh` is unreadable for me right now. This PR merely alleviates that pain. The goal is to not fix all bash scripts now. That needs a larger discussion.

**Comment by colhom** [comment]
@madhusudancs ack. Let's start discussing fixing this bash-formatting problem sooner rather than later- editor settings for shell script formatting are particularly diverse.

**Comment by madhusudancs** [comment]
@colhom sure.

File: federation/cluster/common.sh [hunk 1]
```diff
@@ -139,39 +140,39 @@ function create-federation-api-objects {
...
     done
 
     # Verify that federation-controller-manager pod is running.
-    for i in {1..30};do
-	#TODO(colhom): in the future this needs to scale out for N pods. This assumes just one pod
-	phase="$($host_kubectl get -o=jsonpath pods -lapp=federated-cluster,module=federation-controller-manager --template '{.items[*].status.phase}')"
-	echo "Waiting for federation-controller-manager to be running...(phase= $phase)"
-	if [[ "$phase" == "Running" ]];then
+    for i in {1..30}; do
+	  #TODO(colhom): in the future this needs to scale out for N pods. This assumes just one pod
```
Comment (nikhiljindal) [inline federation/cluster/common.sh]: Add a space after hash here as well

File: federation/cluster/common.sh [hunk 2]
```diff
@@ -139,39 +140,39 @@ function create-federation-api-objects {
 	   create-kubeconfig
 
     # Don't finish provisioning until federation-apiserver pod is running
-    for i in {1..30};do
-	#TODO(colhom): in the future this needs to scale out for N pods. This assumes just one pod
-	phase="$($host_kubectl get -o=jsonpath pods -lapp=federated-cluster,module=federation-apiserver --template '{.items[*].status.phase}')"
-	echo "Waiting for federation-apiserver to be running...(phase= $phase)"
-	if [[ "$phase" == "Running" ]];then
+    for i in {1..30}; do
+      #TODO(colhom): in the future this needs to scale out for N pods. This assumes just one pod
```
Comment (nikhiljindal) [inline federation/cluster/common.sh]: Add a space after hash here as well

**Comment by nikhiljindal** [comment]
Added a couple of suggestions to fix more things.
These changes LGTM

**Comment by goltermann** [comment]
This needs rebase pretty soon if it's going to make 1.3.

**Comment by nikhiljindal** [comment]
Removing from 1.3 as per discussion with @madhusudancs 
(Its code cleanup, no critical functionality)

**Comment by goltermann** [comment]
@madhusudancs this is well out of date.  Needed for v1.4?

**Comment by madhusudancs** [comment]
@goltermann This isn't needed for 1.4. 

I am also working on obsoleting these scripts, so I don't think we will need this. I will close this PR. I will reopen it if necessary.

**Pull request closed without merging**